File: code/smv-debugger/src/test/java/org/fbme/smvDebugger/model/SystemItemParser_Test.java
package org.fbme.smvDebugger.model;

import org.junit.Assert;
import org.junit.Before;
import org.junit.Test;

public class SystemItemParser_Test {
  SystemItemParser itemParser;

  @Test
  public void test_eventPort() {
    final SystemItem item = itemParser.parse("rootFb.fb.event_eventPort");
    Assert.assertEquals(SystemItemType.EVENT_PORT, item.getType());
    Assert.assertArrayEquals(new String[]{"fb"}, item.getFbNames());
    Assert.assertEquals("eventPort", item.getItemName());
  }

  @Test
  public void test_dataPort() {
    final SystemItem item = itemParser.parse("rootFb.fb.varPort");
    Assert.assertEquals(SystemItemType.DATA_PORT, item.getType());
    Assert.assertArrayEquals(new String[]{"fb"}, item.getFbNames());
    Assert.assertEquals("varPort", item.getItemName());
  }

  @Test
  public void test_ecc() {
    final SystemItem item = itemParser.parse("rootFb.fb.Q_smv");
    Assert.assertEquals(SystemItemType.ECC, item.getType());
    Assert.assertArrayEquals(new String[]{"fb"}, item.getFbNames());
    Assert.assertEquals("ECC", item.getItemName());
  }

  @Before
  public void setUp() {
    itemParser = new SystemItemParser();
  }
}

File: code/smv-debugger/src/test/kotlin/org/fbme/smvDebugger/model/SystemItemParser_Test.kt
package org.fbme.smvDebugger.model

import org.junit.Assert
import org.junit.Before
import org.junit.Test

class SystemItemParser_Test {
    var itemParser: SystemItemParser? = null

    @Test
    fun test_eventPort() {
        val item = itemParser!!.parse("rootFb.fb.event_eventPort")
        Assert.assertEquals(SystemItemType.EVENT_PORT, item!!.type)
        Assert.assertArrayEquals(arrayOf("fb"), item.fbNames)
        Assert.assertEquals("eventPort", item.itemName)
    }

    @Test
    fun test_dataPort() {
        val item = itemParser!!.parse("rootFb.fb.varPort")
        Assert.assertEquals(SystemItemType.DATA_PORT, item!!.type)
        Assert.assertArrayEquals(arrayOf("fb"), item.fbNames)
        Assert.assertEquals("varPort", item.itemName)
    }

    @Test
    fun test_ecc() {
        val item = itemParser!!.parse("rootFb.fb.Q_smv")
        Assert.assertEquals(SystemItemType.ECC, item!!.type)
        Assert.assertArrayEquals(arrayOf("fb"), item.fbNames)
        Assert.assertEquals("ECC", item.itemName)
    }

    @Before
    fun setUp() {
        itemParser = SystemItemParser()
    }
}
